adobe_vipm/extension.py

```diff
@@ -3,17 +3,19 @@
 from django.conf import settings
 from ninja import Body
 from swo.mpt.extensions.core import Extension, JWTAuth
+from swo.mpt.extensions.runtime.djapp.conf import get_for_product
 
 from adobe_vipm.flows.fulfillment import fulfill_order
+from adobe_vipm.flows.mpt import get_webhook
 from adobe_vipm.flows.validation import validate_order
 from adobe_vipm.models import Error
 
 ext = Extension()
 
 
-def jwt_secret_callback(claims: Mapping[str, Any]) -> str:
-    secret = settings.EXTENSION_CONFIG["WEBHOOK_SECRET"]
-    return secret
+def jwt_secret_callback(client, claims: Mapping[str, Any]) -> str:
+    product_id = get_webhook(client, claims["webhook_id"])["params"]["product.id"]
```

Comment:
It is `criteria` not `params` according to https://softwareone.atlassian.net/wiki/spaces/GPTEAM/pages/5243142188/Webhook+Object#Fields